cleanup_on_error fails if prepare_workspace succeeds but ingest_data does not

An example: https://github.com/trailofbits/opensearch-benchmark/actions/runs/11564653730/job/32190612550

The `Install AWS CLI` step fails with:
```
[...]
  inflating: aws/dist/docutils/parsers/rst/include/xhtml1-lat1.txt  
  inflating: aws/dist/docutils/parsers/rst/include/isogrk4.txt  
Found preexisting AWS CLI installation: /usr/local/aws-cli/v2/current. Please rerun install script with --update flag.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/benchmark.yml
+++ b/.github/workflows/benchmark.yml
@@ -322,10 +322,10 @@ jobs:
             TF_VAR_benchmark_environment: gh-nightly-${{ needs.prepare_workspace.outputs.now }}
 
   clean_up_on_error:
-    needs: [prepare_workspace, ingest-data]
+    needs: [prepare_workspace]
     runs-on: ubuntu-latest
 
-    if: ${{ always() && contains(needs.*.result, 'failure') }}
+    if: ${{ failure() || cancelled() }}
     
     steps:
       - name: Install AWS CLI

